src/lib.rs

```diff
@@ -273,7 +273,13 @@ pub async fn run_test_case(
             // Verify that each expected key-value pair matches in the response
             expected
                 .iter()
-                .all(|(k, v)| response_json.get(k) == Some(v))
+                .all(|(k, v)| {
+                    if let Ok(json_v) = serde_json::to_value(v) {
+                        response_json.get(k) == Some(&json_v)
+                    } else {
+                        false
```

Comment:
Will coerce the expectation to being "not met" in the event the `serde_json` conversion fails. This may not be ideal.